api/allocations.go

```diff
@@ -162,6 +162,35 @@ func (a AllocIndexSort) Swap(i, j int) {
 	a[i], a[j] = a[j], a[i]
 }
 
+// RescheduleInfo is used to calculate remaining reschedule attempts
+// according to the given time and the task groups reschedule policy
+func (a Allocation) RescheduleInfo(t time.Time) (int, int) {
+	var reschedulePolicy *ReschedulePolicy
+	for _, tg := range a.Job.TaskGroups {
+		if *tg.Name == a.TaskGroup {
+			reschedulePolicy = tg.ReschedulePolicy
+		}
+	}
+	var availableAttempts int
+	var interval time.Duration
+	if reschedulePolicy != nil {
```

Comment:
Would it be the same logic to check if reschedulePolicy is nil or a.RescheduleTracker are nil, and return early instead?